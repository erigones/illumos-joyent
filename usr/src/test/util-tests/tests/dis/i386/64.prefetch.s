/*
 * This file and its contents are supplied under the terms of the
 * Common Development and Distribution License ("CDDL"), version 1.0.
 * You may only use this file in accordance with the terms of version
 * 1.0 of the CDDL.
 *
 * A full copy of the text of the CDDL should have accompanied this
 * source.  A copy of the CDDL is also available via the Internet at
 * http://www.illumos.org/license/CDDL.
 */

/*
 * Copyright 2024 Oxide Computer Company
 */

/*
 * Test prefetch related instructions
 */

.text
.align 16
.globl libdis_test
.type libdis_test, @function
libdis_test:
	prefetch	(%rax)
	prefetcht0	(%r10)
	prefetcht1	(%rbx)
	prefetcht2	0x4(%rcx)
	prefetchw	0x8(%rcx)
	prefetchnta	0x23(%rdx)
	prefetchit0	(%rip)
	prefetchit1	0x18(%rip)
.size libdis_test, [.-libdis_test]
