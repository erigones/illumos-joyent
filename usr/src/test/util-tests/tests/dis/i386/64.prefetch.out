    libdis_test:      0f 0d 00           prefetch (%rax)
    libdis_test+0x3:  41 0f 18 0a        prefetcht0 (%r10)
    libdis_test+0x7:  0f 18 13           prefetcht1 (%rbx)
    libdis_test+0xa:  0f 18 59 04        prefetcht2 0x4(%rcx)
    libdis_test+0xe:  0f 0d 49 08        prefetchw 0x8(%rcx)
    libdis_test+0x12: 0f 18 42 23        prefetchnta 0x23(%rdx)
    libdis_test+0x16: 0f 18 3d 00 00 00  prefetchit0 +0x0(%rip)	<libdis_test+0x1d>
                      00 
    libdis_test+0x1d: 0f 18 35 18 00 00  prefetchit1 +0x18(%rip)	<0x3c>
                      00 
